bin/logstash.lib.sh

```diff
@@ -28,6 +28,26 @@ export LOGSTASH_HOME
 SINCEDB_DIR=${LOGSTASH_HOME}
 export SINCEDB_DIR
 
+# This block will iterate over the command-line args Logstash was started with
+# It will find the argument _after_ --path.settings and use that to attempt
+# to derive the location of an acceptable jvm.options file
+# It will do nothing if this is not found.
+# This fix is for #6379
+if [ -z "$LS_JVM_OPTS" ]; then
+  found=0
+  for i in $@; do
```

Comment:
This should be `"$@"` otherwise it ignores spacing in fun ways that are hard to debug.

Example:

```
# Set 3 arguments. First has a space.
$ set -- "hello world" "two" "three"

# How many args if we don't quote $@?
$ sh -c 'echo $#' - $@
4

# How many if we quote?
$ sh -c 'echo $#' - "$@"
3
```